radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -379,12 +467,646 @@ impl TransactionValidator {
+            return Err(
+                SignatureValidationError::IncorrectNumberOfSubintentSignatureBatches.into(),
+            );
+        }
+
+        let transaction_intent_hash = transaction_intent.transaction_intent_hash();
+
+        let intent_relationships = self.validate_intent_relationships_v2(
+            transaction_intent_hash.into(),
+            &transaction_intent.root_intent_core,
+            subintents,
+        )?;
+
+        self.validate_transaction_header_v2(&transaction_intent.root_header.inner)?;
+
+        let header_aggregation = self.validate_v2_intent_cores(
+            transaction_intent,
+            subintents,
+            &intent_relationships.subintents,
+        )?;
+
+        let transaction_intent_info = ValidatedIntentInformationV2 {
+            children_subintent_indices: intent_relationships.root_intent.children,
+            encoded_instructions: manifest_encode(
+                &transaction_intent.root_intent_core.instructions.inner.0,
+            )?,
+            signature_validations: self.validate_transaction_intent_signatures_v2(&prepared)?,
+        };
+
+        let subintents_info = subintents
+            .iter()
+            .zip(subintent_signatures)
+            .zip(intent_relationships.subintents.into_values())
+            .map(
+                |((subintent, signatures), info)| -> Result<_, TransactionValidationError> {
+                    Ok(ValidatedIntentInformationV2 {
+                        encoded_instructions: manifest_encode(
+                            &subintent.intent_core.instructions.inner.0,
+                        )?,
+                        signature_validations: self
+                            .validate_subintent_signatures_v2(&subintent, &signatures.inner)?,
+                        children_subintent_indices: info.children,
+                    })
+                },
+            )
+            .collect::<Result<_, _>>()?;
+
+        Ok(ValidatedNotarizedTransactionV2 {
+            prepared,
+            overall_epoch_range: EpochRange {
+                start_epoch_inclusive: header_aggregation.overall_start_epoch_inclusive,
+                end_epoch_exclusive: header_aggregation.overall_end_epoch_exclusive,
+            },
+            overall_start_timestamp_inclusive: header_aggregation.overall_start_timestamp_inclusive,
+            overall_end_timestamp_exclusive: header_aggregation.overall_end_timestamp_exclusive,
+            transaction_intent_info,
+            subintents_info,
+        })
+    }
+
+    fn validate_transaction_header_v2(
+        &self,
+        header: &TransactionHeaderV2,
+    ) -> Result<(), HeaderValidationError> {
+        if header.tip_basis_points < self.config.min_tip_basis_points
+            || header.tip_basis_points > self.config.max_tip_basis_points
+        {
+            return Err(HeaderValidationError::InvalidTip);
+        }
+
+        Ok(())
+    }
+
+    fn validate_v2_intent_cores(
```

Comment:
May worth noting that `validate_v2_intent_cores` assumes the transaction has passed `validate_intent_relationships_v2`.

Reply:
True.

It takes a subintent relationship details argument which only comes from validating the relationships - but I'll add a comment clarifying :).